WordPress/gutenberg
# Parser: Remove specific support for <!--more--> tag

## Description
Fixes #2973. This pull request:
- removes the exceptions to the block parser (`` and ``), thus keeping the parser "Gutenberg-focus", as @pento put it;
- treats the More block _more or less_ as any other block, meaning it surrounds the `more` and optional `noteaser` tags with regular comment demarcations;
- retains backwards compatibilty through regular block conversion mechanisms (from freeform via "Convert to Blocks");
- paves the way for implementing a Pagination (``; see #4930) block without introducing further parser exceptions.

## How Has This Been Tested?

Pasting hasn't yet been tested. For conversion of legacy content, try:
1. Using the classic editor, build a post (a lorem ipsum with six paragraphs should suffice). Sprinkle it with the following:

```

```

and

```


```

and

```


```

2. Save it as a draft.
3. Open the post in Gutenberg.
4. Find the freeform (Classic) blocks corresponding to the `more` tags.
5. Convert them to blocks (_advanced block settings > convert to blocks_).
6. Make sure the new More blocks feature the right attributes (`customText` and `noTeaser`). The `customText` attribute can be observed by looking at the text on the More block's visual representation; the `noTeaser` attribute can be observed with the inspector controls.

### Caveat

- `rawHandler`'s transformers will get rid of HTML comments, and will attempt to merge or eliminate many elements (e.g. paragraphs, line breaks). In order to get around this without making special exceptions for certain comments, `commentRemover` will detect `more` and `noteaser` comments and replace them with a custom element `<wp-block>`. This element has no particular definition, is never rendered, and is solely used in order to pass block information through `rawHandler`. The More block's `raw` transform will then match against its shape to replace it with a proper block.
- `commentRemover` happened to be the place where these operations were initially introduced, but the transformer can be split in two.
- The unit tests for the serializer were kept and amended but, as explained in a comment, they should probably be removed. Right now, they illustrate the overall changes of this PR.

## Screenshots (jpeg or gifs if applicable):

## Types of changes





## Checklist:
